Sources/NIOHTTP1Server/main.swift

```diff
@@ -33,6 +33,32 @@ extension String {
     }
 }
 
+private func httpResponseHead(request: HTTPRequestHead, status: HTTPResponseStatus, headers: HTTPHeaders = HTTPHeaders()) -> HTTPResponseHead {
+    var head = HTTPResponseHead(version: request.version, status: status, headers: headers)
+    if request.isKeepAlive && head.headers["connection"].isEmpty && request.version.major == 1 && request.version.minor == 1 {
+        // fast path: HTTP/1.1 with keep-alive and without any 'Connection' response headers pre-set
+        return head
+    }
+
+    let connectionHeaders: [String] = head.headers[canonicalForm: "connection"].map { $0.lowercased() }
```

Comment:
Given that finding a header in the header block is a linear cost operation, I think you may as well request the connection header in canonical form in all cases. That saves you iterating this twice.